joomla/joomla-cms
# [4.0] Improve updater layout

## Summary of Changes
This PR improves the Joomla! updater layout and adds some structure:

### Initial layout
When we're in the initial state and did not check for the update yet, the following screen is visible:

#### Old
![grafik](https://user-images.githubusercontent.com/1229869/125176528-7a884780-e1d4-11eb-9f16-8978c6adb00d.png)

#### New
![grafik](https://user-images.githubusercontent.com/1229869/125176549-925fcb80-e1d4-11eb-82d3-daa353147c9f.png)

As you can see I used the emptystate function, which was implemented a few weeks ago (also in later stages). I also removed the information of the currently installed version, as it gives no benefit in the current state.

Last but not least the "Upload & Update" is moved down in a text link, as I assume upload updates are not that common.

### No update found
Now we press the "Check for updates" without finding an update.

#### Old
![grafik](https://user-images.githubusercontent.com/1229869/125176717-ffc02c00-e1d5-11eb-9f54-0438d6a2314f.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125176725-0babee00-e1d6-11eb-840a-6b8b330e767b.png)

#### New
![grafik](https://user-images.githubusercontent.com/1229869/125176737-28482600-e1d6-11eb-8486-e01ebbc232a7.png)

As you can see, there is no pre-update check, because....there is no update. But the button changes to offer the re-installation of the core files. Also all this unnecessary verbose information were removed.

### Update found
Next test is, if there is an update.

#### Old
![grafik](https://user-images.githubusercontent.com/1229869/125176894-9b05d100-e1d7-11eb-8e14-e986a2650c2d.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125176925-eb7d2e80-e1d7-11eb-8d56-0a18c08dbddd.png)

#### New
![grafik](https://user-images.githubusercontent.com/1229869/125176940-17001900-e1d8-11eb-9a0b-3ea5050a57fb.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125176944-20898100-e1d8-11eb-968f-c4180b509770.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125176948-2f703380-e1d8-11eb-9bee-777a09cb568c.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125176990-9988d880-e1d8-11eb-94ba-f272ec98c941.png)

Here the strange fieldsets are were removed and the information are split into different tab-pills. The update is moved to a 2nd step. Verbose information like current version and the download link are removed (no download needed here, see below in upload area). An additional checkbox is displayed for an extra confirmation regarding backups. Also the confirmation regarding extensions with update required is there.

The pills (left side) show an status icon, for extensions it's an animated hourclass till all extensions are checked.

### Update via upload
Last but not least the update via upload.

#### Old
![grafik](https://user-images.githubusercontent.com/1229869/125177127-9d692a80-e1d9-11eb-9a08-821f1491182e.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125177160-d0abb980-e1d9-11eb-8da4-83120dd830c0.png)

#### New
![grafik](https://user-images.githubusercontent.com/1229869/125177188-051f7580-e1da-11eb-9e82-a8eab2f8600f.png)

and

![grafik](https://user-images.githubusercontent.com/1229869/125177216-38620480-e1da-11eb-8bf1-f1fb3a96d9bb.png)

Here, if a update file is available as download, it's offered to the user to download. Headline of the Login is not "Login" anymore, because the user is already logged in. Also the backup confirmation is here again.

### Update
The update did not change.



### Testing Instructions
It's a bit hard to test. What I did was to install an [Joomla! RC2](https://github.com/joomla/joomla-cms/releases/tag/4.0.0-rc2) and replace the administrator/com_joomlaupdate and media/com_joomlaupdate folders from the download of this PR (see link at the bottom). Also the language file needs to be updated and layouts/joomla/content/emptystate.php has to be copied.

Then you can go through all the steps of the update.

For reinstall, you can use the normal patchtester.

## Actual result BEFORE applying this Pull Request
Update works but layoutit a bit confusing.


## Expected result AFTER applying this Pull Request
Update still works with new layout.

## Additional comment
Help pages has the wrong link.

## TODO
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877710241
- [ ] <s> https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877761125 </s> intended behaviour
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877761600
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877762033
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877762346
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877766531
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877762860
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877765407
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877781674
- [x] https://github.com/joomla/joomla-cms/pull/34754#issuecomment-877714489
